rdflib/util.py

```diff
@@ -203,8 +203,25 @@ def from_n3(s: str, default=None, backend=None, nsm=None):
         return Literal(value, language, datatype)
     elif s == "true" or s == "false":
         return Literal(s == "true")
-    elif s.isdigit():
-        return Literal(int(s))
+    elif (
+        s.lower()
+        .replace('.', '', 1)
+        .replace('-', '', 1)
+        .replace('e', '', 1)
+        .isnumeric()
+    ):
```

Comment:
Couldn't this be better written as:

"""python
if "e" in s.lower():
    return Literal(s, datatype=XSD.double)
else:
    return Literal(float(s), datatype=XSD.decimal) if "." in s else Literal(int(s), datatype=XSD.integer)
"""

Reply:
Changed this as per your suggestion.